frontend/src/todos.js

```diff
@@ -1,10 +1,37 @@
 import React, { useState, useEffect } from 'react';
 import './todos.css';
+import Modal from "./modal";
 import listiclelogoandname from "./images/listiclelogoandname.png";
 // import listiclename from "./images/listiclename.png";
 
 function Todos() {
   const [todos, setTodos] = useState([]);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [formData, setFormData] = useState({ title: "", description: "" });
+
+  const handleAddTodo = (todoData) => {
+    fetch("http://localhost:8000/addtodo", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(todoData),
+    })
+      .then((response) => response.json())
```

Comment:
You can't assume that every response is a good response. You should have different behaviour depending on the kind of response. For example, a successful creation will result in a 200 status code, where as an error has a 500 status code. Fetch will not throw an error on non-200 status codes, so you have to implement that yourself